src/st/st-widget.c

```diff
@@ -382,22 +398,20 @@ st_widget_paint_background (StWidget *widget)
                                     &allocation,
                                     opacity);
   else
-    st_theme_node_paint (theme_node, cogl_get_draw_framebuffer (), &allocation, opacity);
-
-  // ClutterEffect *effect = clutter_actor_get_effect (actor, "background-effect");
-
-  // if (effect == NULL)
-  //   {
-  //     effect = st_background_effect_new ();
-  //     clutter_actor_add_effect_with_name (actor, "background-effect", effect);
-  //   }
-
-  // const char *bumpmap_path = st_theme_node_get_background_bumpmap(theme_node);
-
-  // g_object_set (effect,
-  //               "bumpmap",
-  //               bumpmap_path,
-  //               NULL);
+    {
+      if ((theme_node->background_blur > 0 || theme_node->background_bumpmap)
```

Comment:
Might be good to do these logical comparisons when the change in CSS is detected, instead of on every paint.

Reply:
Perhaps, if I can work out how to do that, but there's only a couple of very fast and cheap tests here, so I can't imagine any practicable difference.

Reply:
i.e. the work to set up the effects is only done the first time through.

Reply:
I see an impact on performance immediately upon using Cinnamon with no CSS changes. You are adding a _lot_ of code to St's paint function. Please try testing in higher resolutions.